its/tests/src/test/java/its/SloopLauncherWithJreTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * SonarLint Core - ITs - Tests
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package its;
+
+import its.utils.PluginLocator;
+import its.utils.SloopDistLocator;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import org.apache.commons.lang3.SystemUtils;
+import org.codehaus.plexus.archiver.UnArchiver;
+import org.codehaus.plexus.archiver.tar.TarGZipUnArchiver;
+import org.codehaus.plexus.archiver.zip.ZipUnArchiver;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.sonarsource.sonarlint.core.rpc.client.SloopLauncher;
+import org.sonarsource.sonarlint.core.rpc.protocol.SonarLintRpcServer;
+import org.sonarsource.sonarlint.core.rpc.protocol.backend.initialize.ClientInfoDto;
+import org.sonarsource.sonarlint.core.rpc.protocol.backend.initialize.FeatureFlagsDto;
+import org.sonarsource.sonarlint.core.rpc.protocol.backend.initialize.InitializeParams;
+
+import static its.SloopLauncherTest.newDummySonarLintClient;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonarsource.sonarlint.core.rpc.protocol.common.Language.GO;
+
+class SloopLauncherWithJreTest {
+  @TempDir
+  private static Path sonarUserHome;
+
+  private static SonarLintRpcServer server;
+
+  @BeforeEach
+  void setup() {
+    var sloopDistPath = SystemUtils.IS_OS_WINDOWS ? SloopDistLocator.getWindowsDistPath() : SloopDistLocator.getLinux64DistPath();
+    var unArchiver = SystemUtils.IS_OS_WINDOWS ? new ZipUnArchiver() : new TarGZipUnArchiver();
+    var outDirPath = unarchiveDistribution(sloopDistPath.toString(), unArchiver);
+    server = SloopLauncher.startSonarLintRpcServerWithJre(sloopDistPath.toString(), newDummySonarLintClient(), outDirPath.resolve("jre").toAbsolutePath().toString());
```

Comment:
If I understand, you are still using the jre provided in the distribution, so this is difficult to confirm the test is actually doing something. Maybe we could rely on the JAVA_HOME being set (it is probably true on our machine and on Cirrus, and run the backend with the JRE located in JAVA_HOME?
Still remain the question of how to assert. Idea: we could log during the backend startup the version/path of the current JRE, and use that to assert?

Reply:
It's correct that the jre provided from the distribution is used but it's being extracted to a different place than where the SloopLanucher will extract it so I would think it's okay to use it. (It's similar to a case where the user has the same version of jre as the one we provide in the distribution.).
I will update the assertion as suggested.

Reply:
Addressed in [55e2773](https://github.com/SonarSource/sonarlint-core/pull/765/commits/55e2773e899579f973ce42cbe69548d564be5dff)